### PR TITLE
Added time subclass

### PR DESCRIPTION
**Pull request by artheist**
Sorry, I said I would separate PRs, but lacking time and I needed the directory structure to test properly.
Added:
- Time proper subcalss of Fraction
- minor python improvements (bool, type checking list)
- added setup.py for installing the module + updated readme
See readme for minimal install instructions, "tested" that it is still playing gabba, works as previously

NOTE: move py dir to py-tidal so that the module has the same name once installed.
Name of the module py_tidal, can be changed in setup.py, but the dir structure needs to follow

### CODE REVIEW

**Comment by yaxu** [comment]
Can we call it `py-vortex` rather than `py-tidal`? I think what we're working on here is not tidal exactly. There might even be a haskell-vortex in the future ;)

File: README.md [hunk 1]
```diff
@@ -1,2 +1,12 @@
 # vortex
 Experiments in tidal ports
+
+To install module, first install pyenv (or conda if you prefer)
+Once in your virtual environment, do:
+
+`cd py-tidal` <br>
```
Comment (yaxu) [inline README.md:7]: change to `py-vortex` or whatever.

Comment (artheist) [reply README.md:7]: Sure, will do, probably later today

File: py-tidal/setup.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+def get_requirements():
+    with open(os.path.join(_base_dir, "requirements.txt")) as fp:
+        return fp.read().splitlines()
+
+setup(
+    name = package_name,
+    version = "0.0.1",
+    description = "Python Tidal port",
+    author = "Tydal Ciclists",
+    author_email = "artheist@lgmail.com",
```
Comment (yaxu) [inline py-tidal/setup.py:20]: I think we can put a list of emails here, or we could make an alias under `@tidalcycles.org` that goes to multiple addresses

Comment (artheist) [reply py-tidal/setup.py:20]: yeah, I was thinking about it, an alias would be best

Comment (artheist) [reply py-tidal/setup.py]: I took vortex@tidalcycles.org hoping that's suitable ...

File: README.md (hunk 1 above)
Comment (artheist) [reply README.md:7]: done

**Comment by yaxu** [comment]
Thanks a lot!

**Pull request merged**